crates/iceberg/src/catalog.rs

```diff
@@ -0,0 +1,163 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Catalog API for Apache Iceberg
+
+use crate::spec::{PartitionSpec, Schema, SortOrder, TableMetadata};
+use crate::Result;
+use async_trait::async_trait;
+use std::collections::HashMap;
+
+/// The catalog API for Iceberg Rust.
+#[async_trait]
+pub trait Catalog {
+    /// Get the catalog configuration for specified warehouse.
+    async fn get_config(&self, warehouse: Option<&str>) -> Result<CatalogConfig>;
+
+    /// List namespaces from table.
+    async fn list_namespaces(&self, parent: Option<NamespaceIdent>) -> Result<Vec<NamespaceIdent>>;
```

Comment:
> All REST clients should first call this route to get catalog configuration
>        properties from the server to configure the catalog and its HTTP client.
>        Configuration from the server consists of two sets of key/value pairs.

According to openapi spec, this should be rest catalog specific call. Maybe we should not put it in `Catalog` api?

Reply:
So catalog doesn't need to have `namespace` concepts?

Reply:
Oh, are you talking about `get_config`?

Reply:
Yes, sorry I  put it in the wrong place.

Reply:
Got it, let's remove it if other catalog doesn't need it.